### PR TITLE
Pub/Sub: Converting Future -> ApiFuture

### PR DESCRIPTION
**Pull request by garrettjonesgoogle**
Also, deleting PullFuture which is subsumed by ApiFuture.

### CODE REVIEW

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/deprecated/spi/PubSubRpc.java [hunk 1]
```diff
@@ -40,189 +41,149 @@
...
-     * @param error the execption that caused the {@link PullFuture} to fail
-     */
-    void failure(Throwable error);
-  }
-
-  /**
-   * A {@link Future} implementation for pull RPCs. This class also allows users to register
-   * callbacks via {@link #addCallback(PullCallback)}.
-   */
-  interface PullFuture extends Future<PullResponse> {
```
Comment (pongad) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/deprecated/spi/PubSubRpc.java]: I'm a little concerned by this. This is definitely a breaking change, even if the fix is pretty obvious. If you think the cost of maintaining it outweighs the cost of the break, I'm good with this.

Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/deprecated/spi/PubSubRpc.java]: This is an "internal" component of Pub/Sub - it's not exposed on the `PubSub` interface, so I'm pretty sure users aren't expected to interact with it.

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/deprecated/PubSubImplTest.java [hunk 2]
```diff
@@ -1407,6 +1381,7 @@ public PullFuture answer() throws Throwable {
     }
   }
 
+
```
Comment (michaelbausor) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/deprecated/PubSubImplTest.java]: Nit: remove newline

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/deprecated/PubSubOptions.java [hunk 3]
```diff
@@ -132,11 +132,6 @@ public Builder toBuilder() {
     return new Builder(this);
   }
 
-  @Deprecated
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/deprecated/PubSubOptions.java]: I was getting an error in IntelliJ - it looks like it was just overlooked.

**Pull request merged**